javascript/javascript2/week3/main/functions.js

```diff
@@ -0,0 +1,392 @@
+          cardsOpened.push(card.id + "1");
+          console.log(cardsOpened);
+          setTimeout(function () {
+            if (cardsOpened[0] === cardsOpened[1]) {
+              if (amountOfCardsOnTheDesk > 2) {
+                amountOfCardsOnTheDesk--;
+                amountOfCardsOnTheDesk--;
+              } else {
+                endGame();
+              }
+              cardsOpened.length = 0;
+              console.log(cardsOpened);
+              const deskCardsToRemove =
+                document.querySelectorAll(".card_flipped");
+              deskCardsToRemove.forEach(function (card) {
+                card.classList.add("card_fade");
+                setTimeout(function () {
+                  deskMenu.removeChild(card);
+                }, 1000);
+              });
+            } else {
+              cardsOpened.length = 0;
+              console.log(cardsOpened);
+              const deskCardsToHide =
+                document.querySelectorAll(".card_flipped");
+              deskCardsToHide.forEach(function (card) {
+                card.className = "card";
+                card.removeAttribute("style");
+              });
+            }
+          }, 1000);
+        } else if (cardsOpened.length < 2) {
+          cardsOpened.push(card.id);
+          console.log(cardsOpened);
+        }
+      }
+    });
+  });
+}
+
+//function, that counts amount of flips
+function countFlipping() {
+  count++;
+  countText.innerText = count;
+}
+
+//function for ending of game
+function endGame() {
+  desk.length = 0;
+  endingMenu.removeAttribute("style");
+  countContainer.setAttribute("style", "display: none;");
+}
+
+//function for restarting the game
+function restartingGame() {
+  count = 0;
+  countText.innerText = count;
+  seconds = 0;
+  minutes = 0;
+  endingMenu.setAttribute("style", "display: none;");
+  startingMenu.removeAttribute("style");
+  countContainer.setAttribute("style", "display: none;");
+}
+
+// //EasterEgg function (I will draw a bunch of cards... Again)
+function setEasterDesk() {
+  const password = document.getElementById("password").value;
+
+  if (password === "3336633") {
+    settingEasterDesk();
+  }
+}
+
+function settingEasterDesk() {
+  count = 0;
+  countText.innerText = count;
+  seconds = 0;
+  minutes = 0;
+  
+  endingMenu.setAttribute("style", "display:none;");
+  easterMenu.setAttribute("style", "display:none;");
+  amountOfCardsOnTheDesk = easterCards.length;
+  //showing container with count
+  countContainer.removeAttribute("style");
+
+  //pushing some amount of cards to desk-array and doubling them (that's why I'm pushing this cards two times)
+  for (let i = 0; i < easterCards.length; i++) {
+    desk.push(easterCards[i]);
+    desk.push(easterCards[i]);
+  }
+
+  //shuffling cards
+  const sortedDesk = desk.sort(() => {
+    return 0.5 - Math.random();
+  });
+
+  //creating cards, setting attributes and setting event listener
+  sortedDesk.forEach((card) => {
```

Comment:
This is the same code as on line 55. Create functions when you find yourself copy-pasting a lot of code.
The comments I wrote before are also applicable here since it is the same code.